core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -641,4 +586,166 @@ public OptionalInt getTraceSamplingPriority(final String authId) {
         }
         return TracingSamplingMode.from(samplingMode).toSamplingPriority();
     }
+
+    /**
+     * Sets the configuration information for this tenant's configured trusted certificate(s).
+     * 
+     * @param configuration A single JSON object property or a list of JSON objects, one set
+     *            of properties for each configured trusted CA.
+     * @return This tenant for command chaining.
+     * 
+     * @throws IllegalArgumentException if configuration is not either a JSON array or JSON Object or
+     *         at least one trusted CA configuration is missing the required <em>subject-dn</em> property.
+     */
+    @SuppressWarnings("unchecked")
+    @JsonProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA)
+    public TenantObject setTrustedCa(final Object configuration) {
+        if (configuration == null) {
+            trustConfigurations = null;
+        } else if (Map.class.isInstance(configuration)) {
+            // for backwards compatibility with the older content model.
+            final JsonObject trustedCa = new JsonObject((Map<String, Object>) configuration);
+            addTrustedCA(trustedCa);
+        } else if (List.class.isInstance(configuration)) {
+
+            final List<Map<String, Object>> configurations = (List<Map<String, Object>>) configuration;
+            configurations.forEach(json -> {
+                final JsonObject trustedCa = new JsonObject(json);
+                addTrustedCA(trustedCa);
+            });
+        } else {
+            throw new IllegalArgumentException("invalid trusted-ca configuration");
+        }
+        return this;
+    }
+
+    /**
+     * Gets the configuration information for this tenant's
+     * configured trusted certificate authorities.
+     * 
+     * @return An unmodifiable list of configuration properties or
+     *         {@code null} if no trust configuration has been
+     *         set for this tenant.
+     */
+    @JsonProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA)
+    public List<Map<String, Object>> getTrustedCa() {
+        if (trustConfigurations == null) {
+            return null;
+        } else {
+            final List<Map<String, Object>> result = new LinkedList<>();
+            trustConfigurations.values().stream().flatMap(Collection::stream).forEach(config -> result.add(config.getMap()));
+            return result;
+        }
+    }
+
+    /**
+     * Get the set of subject distinguished names configure for this tenant.
+     * 
+     * @return The set of subject distinguished names or an empty set if no CA has been configured for this tenant.
+     */
+    @JsonIgnore
+    public Set<X500Principal> getTrustedCaSubjectDns() {
```

Comment:
is this method actually used anywhere?

Reply:
yes, since no two different tenants are allowed to have the same subject name, this method is used to facilitate that check (in `FileBasedTenantService`). however, with this change, a tenant can have multiple trust configs with the same subject name but different public keys